src/aws/credential.rs

```diff
@@ -423,6 +423,10 @@ fn canonicalize_headers(header_map: &HeaderMap) -> (String, String) {
     let mut value_bytes = 0;
     let mut key_bytes = 0;
 
+    fn normalize_whitespace(s: &str) -> String {
+        s.split_whitespace().collect::<Vec<_>>().join(" ")
```

Comment:
The specification only says to convert consecutive spaces, not arbitrary whitespace (although in practice unicode characters or control characters in headers can cause interesting things to happen).